storage/innobase/handler/ha_innodb.cc

```diff
@@ -18915,17 +18923,45 @@ wsrep_abort_transaction(
 	      wsrep_thd_query(victim_thd),
 	      victim_trx ? victim_trx->id : 0);
 
-  if (victim_trx)
+  /* if victim does not have transaction in innodb side,
+     we can use server side abort only */
+  if (victim_trx && victim_trx->state != TRX_STATE_NOT_STARTED)
```

Comment:
The check for `victim_trx->state` is racy here. Trx state is changed to TRX_STATE_ACTIVE without mutex protection in `trx_start_low()`.

Reply:
actually, trx state is not necessarily needed here, checking for id > 0 should be enough to find out if the trx has been registered as RW